### PR TITLE
refactor: clarify all sc-client names

### PR DESCRIPTION
**Pull request by NikitaZotov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,56 +85,56 @@ Create specified in ScConstruction elements.
...
 
 ```ts
     import { client } from "../path-to-client";
 
-    const res = await client.deleteElements([addr1, addr2]);
+    const res = await client.eraseElements([addr1, addr2]);
 ```
 
-## `client.checkElements(addrs: ScAddr[]): ScType[]`
+## `client.getElementTypes(addrs: ScAddr[]): ScType[]`
```
Comment (kilativ-dotcom) [inline README.md]: this name gives vibes that element has several types, what about getElementsTypes?

File: README.md [hunk 2]
```diff
@@ -333,44 +333,44 @@ Generate construction by specified template.
...
 
-    const res = await client.templateGenerate(template, params);
+    const res = await client.generateByTemplate(template, params);
 ```
 
-## `client.eventsCreate(params: ScEventParams[]): ScEvent[]`
+## `client.createElementaryEventSubscriptions(params: ScEventSubscriptionParams[]): ScEventSubscription[]`
 
-Subscribe to event. Event callback, passed to `ScEventParams` constructor, has 4 parameters: subscribedAddr, foundEgde addr, foundNode addr and created eventId. The last one may be used to destroy event after having some specific result.
+Subscribe to event. Event callback, passed to `ScEventSubscriptionParams` constructor, has 4 parameters: subscribedAddr, foundConnector addr, foundNode addr and created eventId. The last one may be used to destroy event after having some specific result.
```
Comment (kilativ-dotcom) [inline README.md]: foundConnector and foundNode? I don't understand what that means

File: README.md [hunk 3]
```diff
@@ -333,44 +333,44 @@ Generate construction by specified template.
...
 
-    const callback = (subscribedAddr: ScAddr, foundEgde: ScAddr, foundNode: ScAddr, createdEventId: number) => {
+    const callback = (subscribedAddr: ScAddr, foundConnector: ScAddr, foundNode: ScAddr, createdEventId: number) => {
         // some logic here
     }
 
-    const evtParams = new ScEventParams(
+    const evtParams = new ScEventSubscriptionParams(
       fakeAddr2,
       ScEventType.RemoveIngoingEdge,
```
Comment (kilativ-dotcom) [inline README.md]: won't event names be corrected?

File: README.md [hunk 4]
```diff
@@ -333,44 +333,44 @@ Generate construction by specified template.
...
     import { client } from "../path-to-client";
 
     const eventIds = [1, 2];
 
-    await client.eventsDestroy(eventIds);
+    await client.destroyElementaryEventSubscriptions(eventIds);
 ```
 
-## `client.findKeynodes(...keynodesInShakeCase: string[]): { keynodesInCamelCase: ScAddr }`
+## `client.searchKeynodes(...keynodesInShakeCase: string[]): { keynodesInCamelCase: ScAddr }`
```
Comment (kilativ-dotcom) [inline README.md]:
```suggestion
## `client.searchKeynodes(...keynodesInSnakeCase: string[]): { keynodesInCamelCase: ScAddr }`
```

File: docs/changelog.md [hunk 5]
```diff
@@ -5,6 +5,54 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Breaking changes
+- This version is compatible with version of the sc-machine 0.10.0. All API methods were redesigned. Misleading ones were removed, new ones were added. See table below, to learn more about changes.
```
Comment (kilativ-dotcom) [inline docs/changelog.md]: don't use 'misleading'

File: docs/changelog.md [hunk 6]
```diff
@@ -5,6 +5,54 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+
+| Deprecated method                      | Substitution method                       | 
+|----------------------------------------|-------------------------------------------|
+| checkElements                          | getElementTypes                           |
+| createElements                         | generateElements                          |
+| createNode                             | generateNode                              |
+| createLink                             | generateLink                              |
+| createEdge                             | generateConnector                         |
+| createElementsBySCs                    | generateElements_by_scs                   |
```
Comment (kilativ-dotcom) [inline docs/changelog.md]: why does it have _by_scs?

File: docs/changelog.md [hunk 7]
```diff
@@ -5,6 +5,54 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+  | Removed sc-event type  | Substitution sc-event type  |
+  |------------------------|-----------------------------|
+  | AddOutgoingEdge        | AfterGenerateOutgoingArc    |
+  | AddIngoingEdge         | AfterGenerateIncomingArc    |
+  | RemoveOutgoingEdge     | BeforeEraseOutgoingArc      |
+  | RemoveIngoingEdge      | BeforeEraseIncomingArc      |
+  | RemoveElement          | BeforeEraseElement          |
+  | ChangeContent          | BeforeChangeLinkContent     |
+
+  New sc-event types: `AfterGenerateConnector`, `AfterGenerateEdge`, `BeforeEraseConnector`, `BeforeEraseEdge` were added.
```
Comment (kilativ-dotcom) [inline docs/changelog.md]:
```suggestion
  - New sc-event types: `AfterGenerateConnector`, `AfterGenerateEdge`, `BeforeEraseConnector`, `BeforeEraseEdge` were added.
```

File: src/ScClient.ts [hunk 8]
```diff
@@ -326,7 +358,15 @@ export class ScClient {
     });
   }
 
-  public async getLinksByContentSubstrings(contents: string[]) {
+  /*!
+   * @deprecated ScClient `getLinksByContents` method is deprecated. Use `searchLinksByContents` instead.
+   */
+  public async getLinksByContents(contents: string[]) {
+    console.warn("Warning: ScClient `getLinksByContents` method is deprecated. Use `getLinksByContents` instead.");
```
Comment (kilativ-dotcom) [inline src/ScClient.ts]:
```suggestion
    console.warn("Warning: ScClient `getLinksByContents` method is deprecated. Use `searchLinksByContents` instead.");
```

File: src/ScConstruction.ts [hunk 9]
```diff
@@ -25,18 +25,26 @@ export class ScConstruction {
...
+  public generateConnector(
     type: ScType,
-    src: string | ScAddr,
-    trg: string | ScAddr,
+    source: string | ScAddr,
+    target: string | ScAddr,
     alias?: string
   ) {
     if (!type.isEdge()) {
       invalidValue("You should pass edge type there");
```
Comment (kilativ-dotcom) [inline src/ScConstruction.ts]: connector?

File: src/ScHelper.ts [hunk 10]
```diff
@@ -90,14 +90,14 @@ export class ScHelper {
...
-    const keynodes = await this._client.findKeynodes(String(addrOrSystemId));
+    const keynodes = await this._client.searchKeynodes(String(addrOrSystemId));
     return keynodes[snakeToCamelCase(String(addrOrSystemId))].value;
   }
 
   public getAnswer(actionNode: ScAddr) {
     return new Promise<ScAddr>((resolve) => {
       (async () => {
-        const { nrelAnswer } = await this._client.findKeynodes("nrel_answer");
+        const { nrelAnswer } = await this._client.searchKeynodes("nrel_answer");
```
Comment (kilativ-dotcom) [inline src/ScHelper.ts]: turns out ts client has answers

File: src/ScSet.ts [hunk 11]
```diff
@@ -31,8 +31,8 @@ export class ScSet {
   private _onInitialize: CallbackInitialize | null = null;
   private _filterType: ScType | null | undefined = null; // just elements with this type will be processed by set
 
-  private _evtAddElement: ScEvent | undefined;
-  private _evtRemoveElement: ScEvent | undefined;
+  private _evtAddElement: ScEventSubscription | undefined;
+  private _evtRemoveElement: ScEventSubscription | undefined;
```
Comment (kilativ-dotcom) [inline src/ScSet.ts]: this line and 69 have 'remove'

File: src/types.ts [hunk 12]
```diff
@@ -179,33 +179,33 @@ type Args<
...
+export type TGetElementsTypesArgs = Args<"check_elements", number[], number[]>;
+export type TEraseElementsArgs = Args<"delete_elements", number[], unknown>;
 export type TKeynodesElementsArgs = Args<
   "keynodes",
   TKeynodesPayload,
   number[]
 >;
-export type TTemplateSearchArgs = Args<
+export type TSearchByTemplateArgs = Args<
   "search_template",
```
Comment (kilativ-dotcom) [inline src/types.ts:190]: what about those strings?

**Review by KovalM — APPROVED** [review]

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (NikitaZotov) [reply README.md]: It is incorrect for English

File: src/types.ts (hunk 12 above)
Comment (NikitaZotov) [reply src/types.ts:190]: These strings are part of SC-JSON. They will be changed in other PR within sc-machine

File: README.md (hunk 1 above)
Comment (kilativ-dotcom) [reply README.md]: getTypesOfElements? getTypes?

File: README.md (hunk 3 above)
Comment (kilativ-dotcom) [reply README.md]: rename them then

File: docs/changelog.md [hunk 13]
```diff
@@ -5,6 +5,54 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+  | RemoveOutgoingEdge     | BeforeEraseOutgoingArc      |
+  | RemoveIngoingEdge      | BeforeEraseIncomingArc      |
+  | RemoveElement          | BeforeEraseElement          |
+  | ChangeContent          | BeforeChangeLinkContent     |
+
+- New sc-event types: `AfterGenerateConnector`, `AfterGenerateEdge`, `BeforeEraseConnector`, `BeforeEraseEdge` were added.
+
+### Added
+- sc-event types: `AfterGenerateConnector`, `AfterGenerateOutgoingArc`, `AfterGenerateIncomingArc`, `AfterGenerateEdge`, `BeforeEraseConnector`, `BeforeEraseOutgoingArc`, `BeforeEraseIncomingArc`, `BeforeEraseEdge`, `BeforeEraseElement`, `BeforeChangeLinkContent`
+- ScClient methods: `getElementTypes`, `generateElements`, `generateNode`, `generateLink`, `generateConnector`, `generateElementsBySCs`, `eraseElements`, `searchLinksByContents`, `searchLinksByContentSubstrings`, `searchLinkContentsByContentSubstrings`, `searchByTemplate`, `generateByTemplate`, `quintuple`, `createElementaryEventSubscriptions`, `destroyElementaryEventSubscriptions`, `searchKeynodes`
```
Comment (kilativ-dotcom) [inline docs/changelog.md]: quintuple is not from ScClient

File: src/ScSet.ts [hunk 14]
```diff
@@ -57,21 +57,21 @@ export class ScSet {
...
-        ScEventType.AddOutgoingEdge,
+        ScEventType.AfterGenerateOutgoingArc,
         this.onEventAddElement.bind(this)
       ),
-      new ScEventParams(
+      new ScEventSubscriptionParams(
         this._addr,
-        ScEventType.RemoveOutgoingEdge,
+        ScEventType.BeforeEraseOutgoingArc,
         this.onEventRemoveElement.bind(this)
```
Comment (kilativ-dotcom) [inline src/ScSet.ts]: addElement and removeElement were renamed in sc-machine

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (NikitaZotov) [reply README.md]: Okay

File: README.md [hunk 15]
```diff
@@ -333,44 +333,44 @@ Generate construction by specified template.
...
-    const callback = (subscribedAddr: ScAddr, foundEgde: ScAddr, foundNode: ScAddr, createdEventId: number) => {
+    const callback = (subscribedAddr: ScAddr, connectorAddr: ScAddr, otherElementAddr: ScAddr, createdEventId: number) => {
         // some logic here
     }
 
-    const evtParams = new ScEventParams(
+    const eventSubscriptionParams = new ScEventSubscriptionParams(
       fakeAddr2,
-      ScEventType.RemoveIngoingEdge,
+      ScEventType.BeforeEraseIngoingArc,
```
Comment (kilativ-dotcom) [inline README.md]: ingoing?

**Review by kilativ-dotcom — APPROVED** [review]

**Pull request merged**